### PR TITLE
Coverage compute orders controller

### PR DESCRIPTION
**Pull request by jvmafra**

### CODE REVIEW

File: src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java [hunk 1]
```diff
@@ -48,132 +48,572 @@
...
 
     private ApplicationFacade facade;
 
-    private final String COMPUTE_ENDPOINT = "/" + ComputeOrdersController.COMPUTE_ENDPOINT;
+    private static final String COMPUTE_ENDPOINT = "/" + ComputeOrdersController.COMPUTE_ENDPOINT;
 
     @Before
     public void setUp() throws FogbowManagerException {
-        this.facade = spy(ApplicationFacade.class);
+        this.facade = Mockito.spy(ApplicationFacade.class);
```
Comment (thiagomanel) [inline src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java]: This exception is never thrown. please, remove it fro the declaration

File: src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java [hunk 2]
```diff
@@ -48,132 +48,572 @@
...
         MvcResult result = this.mockMvc.perform(requestBuilder).andReturn();
-
+        
+        // verify
         int expectedStatus = HttpStatus.BAD_REQUEST.value();
-        assertEquals(expectedStatus, result.getResponse().getStatus());
+        Assert.assertEquals(expectedStatus, result.getResponse().getStatus());
+        Assert.assertEquals("", result.getResponse().getContentAsString());
+        
+        // The request have problems, so the call to Facade is not executed.
```
Comment (thiagomanel) [inline src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java]: has

File: src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java [hunk 3]
```diff
@@ -48,132 +48,572 @@
...
         List<ComputeInstance> computeInstanceList = Arrays.asList(new ComputeInstance[] {computeInstance1, computeInstance2, computeInstance3});
-        doReturn(computeInstanceList).when(this.facade).getAllComputes(anyString());
+        Mockito.doReturn(computeInstanceList).when(this.facade).getAllComputes(Mockito.anyString());
 
         RequestBuilder requestBuilder = createRequestBuilder(HttpMethod.GET, COMPUTE_ENDPOINT, getHttpHeaders(), "");
+        
+        // exercise
         MvcResult result = this.mockMvc.perform(requestBuilder).andReturn();
-
+
```
Comment (thiagomanel) [inline src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java]: extra blank line. remove.

File: src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java [hunk 4]
```diff
@@ -48,132 +48,572 @@
...
+        // exercise
+        MvcResult result = this.mockMvc.perform(requestBuilder).andReturn();
+        
+        // verify
+        int expectedStatus = HttpStatus.OK.value();
+        Assert.assertEquals(expectedStatus, result.getResponse().getStatus());
+        
+        TypeToken<List<InstanceStatus>> token = new TypeToken<List<InstanceStatus>>(){};
+        List<InstanceStatus> resultList = new Gson().fromJson(result.getResponse().getContentAsString(), token.getType());
+        Assert.assertTrue(resultList.size() == 3);
```
Comment (thiagomanel) [inline src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java]: It seems a Assert.assertEquals(3, resultList.size()) fits better

File: src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java [hunk 5]
```diff
@@ -48,132 +48,572 @@
...
+    	final String FAKE_ID = "fake-Id-1";
+        String computeIdEndpoint = COMPUTE_ENDPOINT + "/" + FAKE_ID;
+        Mockito.doThrow(new UnauthenticatedUserException()).when(this.facade).getCompute(Mockito.anyString(), Mockito.anyString());
         RequestBuilder requestBuilder = createRequestBuilder(HttpMethod.GET, computeIdEndpoint, getHttpHeaders(), "");
-
+        
+        // exercise
         MvcResult result = this.mockMvc.perform(requestBuilder).andReturn();
-
+
```
Comment (thiagomanel) [inline src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java]: extra blank line

File: src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java [hunk 6]
```diff
@@ -48,132 +48,572 @@
...
-
+        
+        
+        // verify
+        int expectedStatus = HttpStatus.UNAUTHORIZED.value();
+        Assert.assertEquals(expectedStatus, result.getResponse().getStatus());
+        
+        Mockito.verify(this.facade, Mockito.times(1)).getCompute(Mockito.anyString(), Mockito.anyString());
+    }
+
```
Comment (thiagomanel) [inline src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java]: extra blank line (ask @fgan1 or @arn3tt) for the code formatter they are using.

File: src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java [hunk 7]
```diff
@@ -48,132 +48,572 @@
...
+        // exercise
+        MvcResult result = this.mockMvc.perform(requestBuilder).andReturn();
+        
+        
+        // verify
+        int expectedStatus = HttpStatus.OK.value();
+        Assert.assertEquals(expectedStatus, result.getResponse().getStatus());
 
+        ComputeInstance resultComputeInstance = new Gson().fromJson(result.getResponse().getContentAsString(), ComputeInstance.class);
+        Assert.assertTrue(resultComputeInstance != null);
```
Comment (thiagomanel) [inline src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java]: Assert.assertNotNull fits better

Comment (jvmafra) [reply src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java]: Done

File: src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java (hunk 5 above)
Comment (jvmafra) [reply src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java]: Done

File: src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java (hunk 6 above)
Comment (jvmafra) [reply src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java]: I didn't get it. Blank line between assert and verify?

File: src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java (hunk 4 above)
Comment (jvmafra) [reply src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java]: Done

File: src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java (hunk 3 above)
Comment (jvmafra) [reply src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java]: Done

File: src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java (hunk 2 above)
Comment (jvmafra) [reply src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java]: Done

File: src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java (hunk 1 above)
Comment (jvmafra) [reply src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java]: Done

File: src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java (hunk 6 above)
Comment (thiagomanel) [reply src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java]: 306 307 maybe. check the whole code, there are more dead lines

**Pull request merged**